marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -548,17 +641,37 @@ handleGistAction settings PublishGist = do
                 Just gistId -> runAjax $ flip runReaderT settings $ Server.patchApiGistsByGistId newGist gistId
         assign _createGistResult newResult
         gistId <- hoistMaybe $ preview (_Success <<< gistId) newResult
-        assign _gistId (Just gistId)
-        assign _loadGistResult (Right NotAsked)
+        -- Mark all editors as saved
+        lift do
+          toHaskellEditor $ HaskellEditor.handleAction settings $ HE.MarkProjectAsSaved
+          toJavascriptEditor $ JavascriptEditor.handleAction settings $ JS.MarkProjectAsSaved
+          toSimulation $ Simulation.handleAction settings $ ST.MarkProjectAsSaved
+          void $ query _blocklySlot unit (Blockly.MarkProjectAsSaved unit)
+          void $ query _actusBlocklySlot unit (ActusBlockly.MarkProjectAsSaved unit)
```

Comment:
This is a thought for a separate PR, but there seem to be a lot of places where we say, "Now dispatch this to all the subcomponents." That to me is a sign that something's odd with the data model.

Reply:
I agree, but I'm not sure how to fix it at the moment. I was wondering if this is the time to introduce a Workflow data type and if that would help us (or hurt us) with this problem.

Something like

```purescript
data Workflow 
  = Haskell HaskellState MarloweState BlocklyState
  |  Javascript JavaScriptState MarloweState BlocklyState
  | Basic MarloweState BlocklyState
  | Actus ActusBlocklyState
```

and maybe eventually MarloweState and BlocklyState should be the same so they are always kept on sync?

cc @shmish111